Building the binaries failing.

Disclaimer: I'm relatively new to Go so this may be caused by a lack of understanding.

When I run `make binaries` I get the following error:

```
🐳 bin/swarmd
# _/home/liam/Desktop/swarmkit/cmd/swarmd
cmd/swarmd/main.go:138: cannot use client (type *"github.com/docker/engine-api/client".Client) as type "github.com/docker/swarmkit/vendor/github.com/docker/engine-api/client".APIClient in argument to "github.com/docker/swarmkit/agent/exec/container".NewExecutor:
	*"github.com/docker/engine-api/client".Client does not implement "github.com/docker/swarmkit/vendor/github.com/docker/engine-api/client".APIClient (wrong type for ContainerAttach method)
		have ContainerAttach("golang.org/x/net/context".Context, string, "github.com/docker/engine-api/types".ContainerAttachOptions) ("github.com/docker/engine-api/types".HijackedResponse, error)
		want ContainerAttach("github.com/docker/swarmkit/vendor/golang.org/x/net/context".Context, string, "github.com/docker/swarmkit/vendor/github.com/docker/engine-api/types".ContainerAttachOptions) ("github.com/docker/swarmkit/vendor/github.com/docker/engine-api/types".HijackedResponse, error)
make: *** [bin/swarmd] Error 2
```

Go version is `go version go1.6.2 linux/amd64`

--- a/BUILDING.md
+++ b/BUILDING.md
@@ -1,0 +1,82 @@
+## Build the development environment
+
+To build Swarmkit, you must set up a Go development environment.
+[How to Write Go Code](https://golang.org/doc/code.html) contains full instructions.
+When setup correctly, you should have a GOROOT and GOPATH set in the environment.
+
+After you set up the Go development environment, use `go get` to install `swarmkit`:
+
+	go get github.com/docker/swarmkit/...
+
+This command installs the source repository into the `GOPATH` and builds the
+binaries into `$GOPATH/bin`. Alternatively run the following to download just
+the source:
+
+	go get -d github.com/docker/swarmkit/...
+
+It is not mandatory to use `go get` to checkout the SwarmKit project. However,
+for these instructions to work, you need to check out the project to the
+correct subdirectory of the `GOPATH`: `$GOPATH/src/github.com/docker/swarmkit`.
+
+### Repeatable Builds
+
+For the full development experience, one should `cd` into
+`$GOPATH/src/github.com/docker/swarmkit. From there, the regular `go`
+commands, such as `go test`, should work per package (please see
+[Developing](#developing) if they don't work).
+
+Docker provides a `Makefile` as a convenience to support repeatable builds.
+`make setup` installs tools onto the `GOPATH` for use with developing
+`SwarmKit`:
+
+	make setup
+
+Once these commands are available in the `GOPATH`, run `make` to get a full
+build:
+
+	$ make
+	🐳 fmt
+	🐳 bin/swarmd
+	🐳 bin/swarmctl
+	🐳 bin/swarm-bench
+	🐳 bin/protoc-gen-gogoswarm
+	🐳 binaries
+	🐳 vet
+	🐳 lint
+	🐳 build
+	github.com/docker/swarmkit
+	github.com/docker/swarmkit/vendor/github.com/davecgh/go-spew/spew
+	github.com/docker/swarmkit/vendor/github.com/pmezard/go-difflib/difflib
+	github.com/docker/swarmkit/cmd/protoc-gen-gogoswarm
+	github.com/docker/swarmkit/cmd/swarm-bench
+	github.com/docker/swarmkit/cmd/swarmctl
+	github.com/docker/swarmkit/vendor/github.com/stretchr/testify/assert
+	github.com/docker/swarmkit/ca/testutils
+	github.com/docker/swarmkit/cmd/swarmd
+	github.com/docker/swarmkit/vendor/github.com/pivotal-golang/clock/fakeclock
+	github.com/docker/swarmkit/vendor/github.com/stretchr/testify/require
+	github.com/docker/swarmkit/manager/state/raft/testutils
+	github.com/docker/swarmkit/manager/testcluster
+	github.com/docker/swarmkit/protobuf/plugin/deepcopy/test
+	github.com/docker/swarmkit/protobuf/plugin/raftproxy/test
+	🐳 test
+	?       github.com/docker/swarmkit      [no test files]
+	?       github.com/docker/swarmkit      [no test files]
+	ok      github.com/docker/swarmkit/agent        2.264s
+	ok      github.com/docker/swarmkit/agent/exec   1.055s
+	ok      github.com/docker/swarmkit/agent/exec/container 1.094s
+	?       github.com/docker/swarmkit/api  [no test files]
+	?       github.com/docker/swarmkit/api/duration [no test files]
+	?       github.com/docker/swarmkit/api/timestamp        [no test files]
+	ok      github.com/docker/swarmkit/ca   15.634s
+	...
+	ok      github.com/docker/swarmkit/protobuf/plugin/raftproxy/test       1.084s
+	ok      github.com/docker/swarmkit/protobuf/ptypes      1.025s
+	?       github.com/docker/swarmkit/version      [no test files]
+
+The above provides a repeatable build using the contents of the vendored
+Godeps directory. This includes formatting, vetting, linting, building,
+and testing. The binaries created will be available in `./bin`.
+
+Several `make` targets are provided for common tasks. Please see the `Makefile`
+for details.

--- a/README.md
+++ b/README.md
@@ -31,7 +31,13 @@ Requirements:
 - go 1.6 or higher
 - A [working golang](https://golang.org/doc/code.html) environment
 
-From the project root directory, run:
+*SwarmKit* is built in Go and leverages a standard project structure to work well with Go tooling.
+If you are new to Go, please see [BUILDING.md](BUILDING.md) for a more detailed guide.
+
+Once you have *SwarmKit* checked out in your `$GOPATH`, the `Makefile` can be used for common tasks.
+
+From the project root directory, run the following to build `swarmd` and `swarmctl`:
+
 ```
 make binaries
 ```

